python-checks/src/main/java/org/sonar/python/checks/UseTimeZoneAwareDateTimeConstructorsCheck.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * SonarQube Python Plugin
+ * Copyright (C) 2011-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.python.checks;
+
+import java.util.Set;
+import org.sonar.check.Rule;
+import org.sonar.plugins.python.api.PythonSubscriptionCheck;
+import org.sonar.plugins.python.api.SubscriptionContext;
+import org.sonar.plugins.python.api.symbols.Symbol;
+import org.sonar.plugins.python.api.tree.CallExpression;
+import org.sonar.plugins.python.api.tree.Tree;
+
+@Rule(key = "S6903")
+public class UseTimeZoneAwareDateTimeConstructorsCheck extends PythonSubscriptionCheck {
+  private static final String MESSAGE = "Using timezone aware \"datetime\"s should be preferred over using \"datetime.datetime.utcnow\" and \"datetime.datetime.utcfromtimestamp\"";
```

Comment:
The message should be imperative (sorry it's not part of the specification, it really should be...).
I'd suggest:
Don't use `{FQN}` to create this `datetime` object. 

Also now that I think about it a bit more, I have the feeling that the real message we're trying to convey is that `datetime.datetime.utcnow` and `datetime.datetime.utcfromtimestamp` create timezone-naive objects despite the fact their name imply timezone-aware objects. 

It could very well be that it's perfectly intentional to create a timezone-naive object (and doing otherwise could lead to bugs if compared with other timezone-naive objects). The problem is that the intention is unclear from the method name. Maybe we should discuss that and consider changing the rule title accordingly.